### PR TITLE
Account for <base> tag when calculating integrity.

### LINKED ISSUE #12: Does not take `base` tag into account.
Currently, when processing `tests/index.html` that has:

```
<base href="/">
<script src="assets/vendor.js"></script>
```

We look for `tests/assets/vendor.js`, but we should be looking for `assets/vendor.js`.  This means that no `integrity=` attributes are added for `tests/index.html` (but they are properly added for `index.html`).  This is also causing the ember-cli-sri tests to fail (because none of the `<script>` or `<link>` tags get `integrity=` attributes added).

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes #12.

### CODE REVIEW

**Comment by rwjblue** [comment]
Interestingly the test file contents of `foo/test.html` was not being compared, and it was incorrect (until adding the `<base>` tag as I did here) for exactly the same reason as #12.

**Comment by stefanpenner** [comment]
Lgtm. I assume this was an existing bug, not related to my work this morning.

File: index.js [hunk 1]
```diff
@@ -55,10 +56,26 @@ function SRIHashAssets(inputNodes, options) {
...
 
+SRIHashAssets.prototype.getBaseHREF = function getBaseHREF(string) {
+  var baseTag = string.match(BASE_CHECK);
+  if (baseTag && baseTag[0]) {
+    var href = baseTag[0].match(HREF_CHECK);
+    var relativePath = href && href[1];
+
+    if (!relativePath) { return null; }
+
+    return this.inputPaths[0] + relativePath;
```
Comment (stefanpenner) [inline index.js:70]: should we worry about '../../../` basePaths? (ones that escape out of the root)

Comment (rwjblue) [inline index.js:70]: Hmm. I'm not sure? I haven't really seen a relative base url like that, is it valid?

**Comment by rwjblue** [comment]
@stefanpenner - Yes, it was an existing issue (though in 1.1.1 it assumed _everything_ was relative to / so it was somewhat the reverse problem).

**Comment by jonathanKingston** [comment]
Personally I would prefer to turn off integrity checking in that instance just for paranoias sake really.
In fact I would argue anything other than `'/'` could do the same which supplies a nice carrot/stick for people to move over to using a nice base tag. (That is however me, I am cruel etc)

**Comment by jonathanKingston** [comment]
Also again thank you so much for your hard work comrades.

File: index.js (hunk 1 above)
Comment (rwjblue) [inline index.js:70]: Ya, just confirmed (via [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)) that it can be relative or absolute.  

I agree with @jonathanKingston, we can opt out of SRI if `<base>` resolves to a path above the project root.

Comment (rwjblue) [inline index.js:70]: OK, I added a guard to ensure that we do not use a `<base>` that doesn't start with `/` (which should prevent escaping out).
